index.js

```diff
@@ -48,21 +57,18 @@ module.exports = function(namespace) {
  * @param {string} namespaces
  */
 module.exports.setNamespaces = function(namespaces) {
-    internals.namespaces = namespaces
-    internals.enabledNamespaces = []
-    internals.disabledNamespaces = []
+    exports.namespaces = namespaces
+    internals.namespaces = []
+
+    if (_.isEmpty(namespaces)) return internals.syncLoggers()
 
-    namespaces = _.split(namespaces || '', /[\s,]+/)
+    namespaces = _.split(_.trim(namespaces) || '', /[\s,]+/)
 
     _.forEach(namespaces, namespace => {
-        if (_.isEmpty(namespace)) return true
+        const parsedNamespace = internals.parseNamespace(namespace) || {}
+        if (!parsedNamespace) return true
```

Comment:
As you provide a default `|| {}` and `{}` is truthy, this return statement will never be reached.